WooCommerce/Classes/ViewRelated/ReusableViews/TitleAndEditableValueTableViewCell/TitleAndEditableValueTableViewCell.swift

```diff
@@ -2,10 +2,10 @@ import UIKit
 
 final class TitleAndEditableValueTableViewCell: UITableViewCell {
     @IBOutlet private weak var stackView: UIStackView!
-    // TODO Make this private and use `update()` instead.
-    @IBOutlet weak var title: UILabel!
-    // TODO Make this private and use `update()` instead.
-    @IBOutlet weak var value: UITextField!
+    @IBOutlet private weak var title: UILabel!
+    @IBOutlet private weak var value: UITextField!
+
+    private var viewModel: TitleAndEditableValueTableViewCellViewModel?
```

Comment:
Interesting... 🤔  
I normally don't wire "models" to a view because we end up with two sources of truth. The model state and the UIKit state. Additionally, this also increases complexity in views which I considered to be dumb. 

I understand how it's more convenient to get the value every time the field is updated rather than fetch it through the view hierarchy when the action button is pressed. But still, It feels riskier to me. 

Am I worrying about nothing? 😛

Reply:
> I normally don't wire "models" to a view because we end up with two sources of truth. The model state and the UIKit state.

In this case, I was actually aiming for the `ViewModel` to be the only source of truth. If you look at `self.update()`, once a `ViewModel` is set, the View is reset to whatever the `ViewModel`'s _state_ is. So I believe this View is still dumb. 

I think the complexity here is because we have to use `UITableView` and `UITableView` requires us to make the View (Cells) **reusable**. If there was a way to bind 1 `Cell` to 1 `ViewModel` **forever** then I would opt for that. But I don't know if that's possible right now. 

> But still, It feels riskier to me.

I'm curious about the risks! 

I was aiming for this to be reusable everywhere else. What I noticed in our app is that we use a lot of UI components and they are mostly very similar (especially in Products). My vision is for these UI components to be reusable and not tightly bound to the ViewModel. And also that we don't need to write a lot of code in order to bind them together. There's also an opportunity for us to have one `ViewModel` but different views for iPhone and iPad. 

I don't really know how it's gonna look like yet. I started proposing this for Product Settings (diagrams and discussions here https://github.com/woocommerce/woocommerce-ios/pull/1854#pullrequestreview-357680040) but I'm waiting for iOS 13 to start looking at a design proposal.

Reply:
> In this case, I was actually aiming for the ViewModel to be the only source of truth. If you look at self.update(), once a ViewModel is set, the View is reset to whatever the ViewModel's state is. So I believe this View is still dumb.

Hmm, thinking a bit more about it, I guess this is a fancy(or more complex) way of expressing a delegate.

> I'm curious about the risks!

The risk that I fear is that somehow the UI gets updated but the view fails to transmit the value to the view model, so we end up with two sources of truth.

Reply:
> The risk that I fear is that somehow the UI gets updated but the view fails to transmit the value to the view model, so we end up with two sources of truth.

That makes sense! I added a test for that but it's certainly possible.